clock/src/main/java/io/deephaven/base/clock/ClockNanoBase.java

```diff
@@ -0,0 +1,36 @@
+/**
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+package io.deephaven.base.clock;
+
+import java.time.Instant;
+
+/**
+ * A base implementation of Clock, with all methods being sourced from {@link #currentTimeNanos()}.
+ */
+public abstract class ClockNanoBase implements Clock {
+    @Override
+    public final long currentTimeMillis() {
+        return currentTimeNanos() / 1_000_000;
+    }
+
+    @Override
+    public final long currentTimeMicros() {
+        return currentTimeNanos() / 1_000;
+    }
+
+    @Override
+    public final Instant instantNanos() {
+        return Instant.ofEpochSecond(0, currentTimeNanos());
+    }
+
+    @Override
+    public final Instant instantMillis() {
+        return Instant.ofEpochSecond(0, currentTimeNanos());
+    }
+
+    @Override
+    public final long nanoTime() {
+        return currentTimeNanos();
```

Comment:
This is a larger question of "Should we have unit-testability for nanoTime logic; and should that interface be on Clock?"

Confirmed w/ Charles, we might want fine replayability of nanoTime for unit tests, even though we don't have them right now.

The next part is "should this be part of clock?"

It _may_ be reasonable to have a separate interface since it is more bespoke than Clock; but what I *don't* want to encourage is users just deciding to use `Clock#currentTimeNanos` because it's more convenient than plumbing through an additional nanoTime interface. That said, there aren't many users of `Clock#nanoTime` (only AutoTuningIncrementalReleaseFilter and BaseIncrementalReleaseFilter), so it would be a small change against this PR.

Options:
1. Leave `Clock#nanoTime` as-is
2. Create new interface `MeasureNanos`.
3. Remove `Clock#nanoTime`, and explicitly use `System.nanoTime` in BaseIncrementalReleaseFilter/AutoTuningIncrementalReleaseFilter

I'm actually leaning 3 > 2 > 1; at least until we get to the point where we _need_ nanoTime replayability for unit tests.

Reply:
I kind of like #3 for now. It feels better not to conflate sources of wall-clock time with sources of monotonically-increasing measurement time. I'm not going to complain too much if we do #2 or #1, though, at this point.

Reply:
Proceeded w/ 3